### PR TITLE
feat: Games and CardTypes CRUD

### PR DESCRIPTION
**Pull request by xalluna**

### CODE REVIEW

File: TcgPocket/Features/Games/GamesController.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+    
+    [HttpPost]
+    public async Task<ActionResult<Response<GameGetDto>>> CreateGame([FromBody] GameDto data)
+    {
+        var response = await _mediator.Send(new CreateGameRequest{Game = data});
+        
+        return response.HasErrors ? BadRequest(response) : CreatedAtRoute(nameof(CreateGameRequest), response);
```
Comment (gbrad777) [inline TcgPocket/Features/Games/GamesController.cs:41]: add id part into route *see other created comment above*

File: TcgPocket/Features/Games/Queries/GetAllGamesRequest.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using TcgPocket.Data;
+using TcgPocket.Shared;
+
+namespace TcgPocket.Features.Games.Queries;
+
+public class GetAllGamesRequest : IRequest<Response<List<GameDto>>>
+{
+    public int Page { get; set; }
```
Comment (gbrad777) [inline TcgPocket/Features/Games/Queries/GetAllGamesRequest.cs:12]: same question as the last time. Overkill?

File: TcgPocket/Features/Games/Queries/GetGameByIdRequest.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using AutoMapper;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using TcgPocket.Data;
+using TcgPocket.Shared;
+
+namespace TcgPocket.Features.Games.Queries;
+
+public class GetGameByIdRequest : IRequest<Response<GameDto>>
```
Comment (gbrad777) [inline TcgPocket/Features/Games/Queries/GetGameByIdRequest.cs:9]: validate Id from request?

File: TcgPocket/Shared/Enums.cs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿namespace TcgPocket.Features;
+﻿namespace TcgPocket.Shared;
```
Comment (gbrad777) [inline TcgPocket/Shared/Enums.cs:2]: do we even need this? are we just trying to avoid putting the "DefaultConnection" bit in the startup?

Comment (xalluna) [reply TcgPocket/Shared/Enums.cs:2]: I don't want to have to change every instance of 'DefaultConnection' if something needs to change. This is a convention I would like to follow for all app settings that get used. Tracking down string literals is a nightmare and this reduces that surface area.

File: TcgPocket/Features/CardTypes/Validators/CreateCardTypeRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+using FluentValidation;
+using TcgPocket.Features.CardTypes.Commands;
+
+namespace TcgPocket.Features.CardTypes.Validators;
+
+public class CreateCardTypeRequestValidator : AbstractValidator<CreateCardTypeRequest>
+{
```
[the remark this answers is not in the document]
Comment (xalluna) [reply TcgPocket/Features/CardTypes/Validators/CreateCardTypeRequestValidator.cs:7]: I don't understand how this breaks dependency injection and I don't think it's necessary.

File: TcgPocket/Features/CardTypes/Commands/CreateCardTypeRequest.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+using TcgPocket.Shared;
+
+namespace TcgPocket.Features.CardTypes.Commands;
+
+public class CreateCardTypeRequest : IRequest<Response<CardTypeGetDto>>
+{
+    public CardTypeDto CardType { get; set; }
+}
+
+public class CreateCardTypeRequestHandler : IRequestHandler<CreateCardTypeRequest, Response<CardTypeGetDto>>
```
[the remark this answers is not in the document]
Comment (xalluna) [reply TcgPocket/Features/CardTypes/Commands/CreateCardTypeRequest.cs:16]: I could have just been tired, but I thought I was doing this already and we specifically didn't want this. I can change it back tho

File: TcgPocket/WeatherForecast.cs
Comment (DeannaShort) [inline TcgPocket/WeatherForecast.cs:1]: How could you? 😢

File: TcgPocket/Shared/Enums.cs (hunk 4 above)
Comment (gbrad777) [reply TcgPocket/Shared/Enums.cs:2]: 👌

File: TcgPocket/Features/Games/GameMapper.cs [hunk 7]
```diff
@@ -9,6 +9,6 @@ public GameMapper()
     {
         CreateMap<Game, GameGetDto>();
         CreateMap<Game, GameDto>().ReverseMap();
-        CreateMap<CreateGameRequest, Game>().ReverseMap();
+        CreateMap<CreateGameCommand, Game>().ReverseMap();
```
Comment (gbrad777) [inline TcgPocket/Features/Games/GameMapper.cs]: remove dis boi, game to gamedto does this for us

**Review by gbrad777 — APPROVED** [review]

**Pull request merged**